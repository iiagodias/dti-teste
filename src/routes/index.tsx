import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import MainStack from './MainStack';

const routes: React.FC = () => {
  return (
    <NavigationContainer>
      <MainStack />
    </NavigationContainer>
  );
}

export default routes;