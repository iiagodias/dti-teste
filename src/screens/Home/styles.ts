import styled from 'styled-components/native';

export const Container = styled.SafeAreaView`
  flex: 1;
  background-color: ${({ theme }) => theme.background};
`;

export const Scroll = styled.ScrollView``;


export const Body = styled.View`
  padding: 15px;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;
`;

export const ContainerOrder = styled.View`
  flex-direction: row;
  justify-content: space-between;
  padding: 15px;
`;

export const ContainerOrderButton = styled.TouchableOpacity``;

export const TypeOrder = styled.Text`
  font-size: 16px;
  color: ${({ theme }) => theme.secondary};
`;

export const Orderby = styled.Text`
 font-size: 16px;
 color: ${({ theme }) => theme.secondary};
`;
