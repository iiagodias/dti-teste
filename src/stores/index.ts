
import { applyMiddleware, createStore, Store } from 'redux';
import createSagaMiddleware from 'redux-saga';
import reducers from './ducks';
import { IMoviesState } from './ducks/movies/types';

export interface IAplicationStates {
  movies: IMoviesState
}

const sagaMiddleware = createSagaMiddleware();


const store: Store<IAplicationStates> = createStore(reducers, applyMiddleware(sagaMiddleware));


export { store };
