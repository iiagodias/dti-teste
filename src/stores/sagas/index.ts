import { all } from 'redux-saga/effects';
import movieWatcher from './movies';


function* rootSagas() {
  yield all([movieWatcher()]);
}

export default rootSagas;