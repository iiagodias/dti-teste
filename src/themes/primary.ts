export default {
  primary: '#f1c40f',
  secondary: '#191919'
}