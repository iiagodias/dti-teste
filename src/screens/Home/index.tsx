import React from 'react';
import SearchBox from '../../components/SearchBox';
import { Container } from './styles';

const Home: React.FC = () => {
  return (
    <Container>
      <SearchBox />
    </Container>
  );
}

export default Home;