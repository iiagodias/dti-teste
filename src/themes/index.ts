import 'styled-components';
import primary from './primary';

type Primary = typeof primary

declare module 'styled-components' {
  export interface DefaultTheme extends Primary { }
}

export { primary };
