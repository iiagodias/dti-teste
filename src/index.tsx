import React from 'react';
import 'react-native-gesture-handler';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import Routes from './routes';
import { store } from './stores';
import { primary } from './themes';

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={primary}>
        <Routes/>
      </ThemeProvider>
    </Provider>
  );
}

export default App;