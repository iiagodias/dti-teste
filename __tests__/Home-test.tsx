jest.useFakeTimers()

import { render } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import Home from '../src/screens/Home';
import { store } from '../src/stores';
import { primary } from '../src/themes';


const AllTheProviders: React.FC = ({ children }) => {
  return (
    <Provider store={store}>
        <ThemeProvider theme={primary}>
          {children}
        </ThemeProvider>
    </Provider>
  )
}


 it('Theme styled component and redux', async () => {
    const {toJSON} = render(<Home />, {wrapper: AllTheProviders});

    expect(toJSON()).toMatchSnapshot()
 });