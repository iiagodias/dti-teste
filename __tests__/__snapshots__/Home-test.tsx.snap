// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme styled component and redux 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ecf0f1",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f1c40f",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Digite o nome"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#ecf0f1",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#191919",
            "height": 45,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": "50%",
          },
        ]
      }
      testID="input-title"
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      options={
        Object {
          "format": "YYYY",
        }
      }
      placeholder="Ano"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#ecf0f1",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#191919",
            "height": 45,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": "30%",
          },
        ]
      }
      type="datetime"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="button-submit"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#191919",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#191919",
              "fontSize": 16,
            },
          ]
        }
      >
        Organizar por titúlo
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#191919",
              "fontSize": 16,
            },
          ]
        }
      >
        A - z
      </Text>
    </View>
  </View>
  <RCTScrollView
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            },
          ]
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
